x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -405,19 +397,45 @@ public void testStringAsLookupIndexPattern() {
         assertStringAsLookupIndexPattern("cluster*:index*", "ROW x = 1 | LOOKUP cluster*:index* ON j");
         assertStringAsLookupIndexPattern("cluster*:*", "ROW x = 1 | LOOKUP cluster*:* ON j");
         assertStringAsLookupIndexPattern("*:index*", "ROW x = 1 | LOOKUP  *:index* ON j");
         assertStringAsLookupIndexPattern("*:*", "ROW x = 1 | LOOKUP  *:* ON j");
+    }
 
+    public void testInvalidCharacterInIndexPattern() {
+        for (String command : List.of("FROM {}", "METRICS {}", "ROW x = 1 | LOOKUP {} ON j")) {
+            expectError(command, " cluster:\"index|pattern\"", "Invalid index name [index|pattern]");
+            expectError(command, " *:\"index|pattern\"", "Invalid index name [index|pattern]");
+            expectError(command, " cluster:\"index#pattern\"", "Invalid index name [index#pattern]");
+            expectError(command, " cluster:index#pattern", "Invalid index name [index#pattern]");
+            expectError(command, " cluster:\"index pattern\"", "Invalid index name [index pattern]");
+            expectError(command, " cluster:\"index?pattern\"", "Invalid index name [index?pattern]");
+            expectError(command, " cluster:index?pattern", "Invalid index name [index?pattern]");
+            expectError(command, " cluster:\"index>pattern\"", "Invalid index name [index>pattern]");
+            expectError(command, " cluster:index>pattern", "Invalid index name [index>pattern]");
+            expectError(command, " cluster:\"index<pattern\"", "Invalid index name [index<pattern]");
+            expectError(command, " cluster:index<pattern", "Invalid index name [index<pattern]");
+            expectError(command, " cluster:\"index/pattern\"", "Invalid index name [index/pattern]");
+            expectError(command, " cluster:index/pattern", "Invalid index name [index/pattern]");
+            expectError(command, " cluster:\"index\\\\pattern\"", "Invalid index name [index\\pattern]");
+            expectError(command, " cluster:index\\pattern", "Invalid index name [index\\pattern]");
+            expectError(command, " cluster:\"..\"", "Invalid index name [..]");
+            expectError(command, " cluster:..", "Invalid index name [..]");
+            expectError(command, " cluster:\"_indexpattern\"", "Invalid index name [_indexpattern]");
+            expectError(command, " cluster:_indexpattern", "Invalid index name [_indexpattern]");
+            expectError(command, " cluster:\"+indexpattern\"", "Invalid index name [+indexpattern]");
+            expectError(command, " cluster:+indexpattern", "Invalid index name [+indexpattern]");
+            expectError(command, " cluster:\"--indexpattern\"", "Invalid index name [-indexpattern]");
+            expectError(command, " cluster:--indexpattern", "Invalid index name [-indexpattern]");
+            expectError(command, " cluster:\"<--logstash-{now/M{yyyy.MM}}>\"", "Invalid index name [-logstash-");
```

Comment:
Interesting that the failure is about `-logstash-` and wondering if it's what we want.
Right now a `from \"<--logstash-{now/M{yyyy.MM}}>\"` (actually, w/ or w/o the quotes) yields: `Unknown index [<--logstash-{now/M{yyyy.MM}}>]` which _seems_ more correct?

Reply:
Indeed, it is the exclusion that gives us some inconsistent behaviors. If there is an index name with wildcard that matches existing index names, the invalid index names starting with exclusions are ignored. Otherwise if there is no existing index name that matches the wildcard, there is `VerificationException` for Unknown index, however `<--logstash-{now/M{yyyy.MM}}>` is indeed an invalid index name.

One option that I can think of is that, we skip the validation of index names that start with exclusions in parser, because at that time, we couldn't validate whether there is any existing index name that matches the wildcard or not. The validation of index names starting with exclusions is deferred to later phases. This is less inconsistent with the current behavior, but it might miss some invalid index names.